src/main/java/org/opensearch/jdbc/ConnectionImpl.java

```diff
@@ -83,8 +87,11 @@ log, new SQLNonTransientException("Could not initialize transport for the connec
             ConnectionResponse connectionResponse = this.protocol.connect(connectionConfig.getLoginTimeout() * 1000);
             this.clusterMetadata = connectionResponse.getClusterMetadata();
             this.open = true;
+        } catch (HttpException ex) {
+            logAndThrowSQLException(log, new SQLException("Connection error " + ex.getMessage(),
```

Comment:
You don't have check for 401 now.

Reply:
Please take a look at the expected error message and make sure we're aligned.  I don't think "Connection Error" needs to be included.

Reply:
I think we still want "Connection Error" to give non-Tableau users an error message or else they will only get a SQLSTATE.

Reply:
@Yury-Fridlyand I'm not sure if we want to check for 401 since it doesn't cover AWS_SIGV4 invalid credentials. I think we need to return 28000 for invalid credentials and not only for BASIC auth according to [Tableau driver specifications](https://tableau.github.io/connector-plugin-sdk/docs/drivers#:~:text=Tableau%20expects%20correct%20SQLStates%20to%20be%20returned%20in%20certain%20situations.%20Inaccurate%20SQLStates%20(for%20example%2C%20returning%20a%20generic%20error%20instead%20of%20SLQState%2028000%20for%20invalid%20credentials)%20may%20cause%20bugs%20in%20the%20connector).

Reply:
Right. Still consider adding check for status 401.

Reply:
Thanks. I reverted to check for 401 as it is only for authorization. Also added 08001 state to catch other causes for failure to connect. Changed in https://github.com/Bit-Quill/sql-jdbc/pull/7/commits/b6ba2cda980c161bb32dba67190463e0edb053b2